---
title: "Hw 8 - Data wrangling for genomics"
format: docx
editor: visual
---

Fill out [this Quarto document](https://github.com/BIFX552-22/CourseInfo/hw/08wrangling.qmd) while completing the lesson on data wrangling for genomics.

## [Background and Metadata](https://datacarpentry.org/wrangling-genomics/01-background/index.html)

Rather than copy the 1+ GB directory with the data for this lesson a bunch of times (once for each person), we'll create a symbolic link to the data and copy over the bits that we will be modifying ourselves. Navigate to your home directory and do the following:

```{{bash}}
# go home
cd ~

# create a new `shell_data` directory
mkdir Ecoli
cd Ecoli

# create directories that we will be working with
mkdir results
mkdir scripts
mkdir trimmed_fastq

# link new data directories to your directory
ln -s /BIODATA/BIFX552/Ecoli/Ecoli_metadata_composite.csv .
ln -s /BIODATA/BIFX552/Ecoli/data .
ln -s /BIODATA/BIFX552/Ecoli/untrimmed_fastq .
```

If you want to work locally (e.g. on your laptop), you can follow the instructions [here](https://datacarpentry.org/genomics-workshop/setup.html) to get your environment set up.

### Challenge 1

Based on the metadata, can you answer the following questions?

* How many different generations exist in the data?
* How many rows and how many columns are in this data?
* How many citrate+ mutants have been recorded in Ara-3?
* How many hypermutable mutants have been recorded in Ara-3?


## [Quality Control](https://datacarpentry.org/wrangling-genomics/02-quality-control/index.html)

### Challenge 2

* What is the last read in the SRR2584863_1.fastq file?
* How confident are you in this read?

### Challenge 3

How big are the files in `untrimmed_fastq/` (show code and output)?

```{{bash}}
# insert code here
```

### Challenge 4

After running `fastqc` on your untrimmed data, which sample(s) looks the best in terms of per base sequence quality? Which sample(s) look the worst?

### Challenge 5

Which samples failed at least one of FastQCâ€™s quality tests? What test(s) did those samples fail?


## [Trimming and Filtering](https://datacarpentry.org/wrangling-genomics/03-trimming/index.html)

### Challenge 6

Use the output from your Trimmomatic command to answer the following questions.

* What percent of reads did we discard from our sample?
* What percent of reads did we keep both pairs?

### Challenge 7

We trimmed our fastq files with Nextera adapters, but there are other adapters that are commonly used. What other adapter files came with Trimmomatic?

### Challenge 8

Now that our samples have gone through quality control, they should perform better on the quality tests run by FastQC. Go ahead and re-run FastQC on your trimmed FASTQ files and visualize the HTML files to see whether your per base sequence quality is higher after trimming.


## [Variant Calling](https://datacarpentry.org/wrangling-genomics/04-variant_calling/index.html)

### Challenge 9
We saved the reference genome as `data/ecoli_rel606.fasta.gz` and then decompressed it. What is the real name of the genome?

### Challenge 10

Use the `grep` and `wc` commands you have learned to assess how many variants are in the vcf file.

### Challenge 11

Visualize the alignment of the reads for our `SRR2584863` sample. What variant is present at position 433359? What is the canonical nucleotide in that position?
